source/base/mpi_remote_point_evaluation.cc

```diff
@@ -207,7 +207,7 @@ namespace Utilities
       const unsigned int i) const
     {
       AssertIndexRange(i, point_ptrs.size());
-      return (point_ptrs[i] - point_ptrs[i - 1]) > 0;
+      return (point_ptrs[i] - (i > 0) ? point_ptrs[i - 1] : 0) > 0;
```

Comment:
```suggestion
      return (point_ptrs[i+1] - point_ptrs[i]) > 0;
```

Reply:
Just to be sure: Is it then correct that `all_points_found()` begins the loop with `i=1`?

Reply:
Yes it is right.

Reply:
And the AssertIndexRange ??

Reply:
`AssertIndexRange(i, point_ptrs.size() - 1);`

Reply:
But then the loop in `all_points_found()` can not run its last element, right?

Reply:
Now I understand what you mean! `all_points_found()` calls this function. It has to loop from 0 to `point_ptrs.size() - 1`. Please add an if-statement around the loop `if(point_ptrs.size() > 0)`.